### PR TITLE
Replace TravisCI with Github Actions

### PR DESCRIPTION
**Pull request by Christopher-Tom**
First try with Evan :)

### CODE REVIEW

**Comment by evanugarte** [comment]
this is a work in progress, we arent done yet

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: python setup.py
+    - run: npm ci
+    - run: cd api
+    - run: npm ci
```
Comment (evanugarte) [inline .github/workflows/node.js.yml:31]:
```suggestion
    - run: npm ci
    - run: cd ..
```

**Review by evanugarte — CHANGES_REQUESTED** [review]

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
```
Comment (evanugarte) [inline .github/workflows/node.js.yml:2]: remove these comments, make the first line be what's on line 4 right now.

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+name: Node.js CI
```
Comment (evanugarte) [inline .github/workflows/node.js.yml:4]:
```suggestion
name: Core-v4 Tests
```

File: .github/workflows/node.js.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+on:
+  push:
+    branches: [ dev ]
```
Comment (evanugarte) [inline .github/workflows/node.js.yml:8]:
```suggestion
```

Comment (evanugarte) [reply .github/workflows/node.js.yml:8]: Lets remove this so the tests only run when a pr is open.

File: .github/workflows/node.js.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  pull_request:
+    branches: [ dev ]
+
+jobs:
+  build:
+
```
Comment (evanugarte) [inline .github/workflows/node.js.yml:14]:
```suggestion
```

File: .github/workflows/node.js.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    runs-on: ubuntu-latest
+
```
Comment (evanugarte) [inline .github/workflows/node.js.yml:16]:
```suggestion
```

File: .github/workflows/node.js.yml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    strategy:
+      matrix:
+        node-version: [12.x]
+        mongodb-version: [4.2]
+
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+
```
Comment (evanugarte) [inline .github/workflows/node.js.yml:23]: remove these lines

**Review by evanugarte — CHANGES_REQUESTED** [review]
Looks good! See comments

**Review by evanugarte — APPROVED** [review]
talk to em 🗣️

**Review by hsorathia — APPROVED** [review]

**Review by seemavora — APPROVED** [review]

**Review by jerrylee17 — APPROVED** [review]
I'd prefer keeping `.travis.yml` in case we can use travis again... Just make a note saying it does not work at the moment.

**Comment by IsitaB** [comment]
Shouldn't we add `windows-latest` and `mac-latest` to the os list? Is ubuntu enough?

Which version of node does `matrix` pull? Can we change this to pull the node version from package.json?

**Review by law-nguyen — APPROVED** [review]
I too am curious about isita's comment, why only ubuntu?

File: .github/workflows/node.js.yml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Starting MongoDB
+      uses: supercharge/mongodb-github-action@1.3.0
+    - name: Use Node.js ${{ matrix.node }}
```
Comment (charlotte-zhuang) [inline .github/workflows/node.js.yml:18]:
```suggestion
    - name: Use Node.js
```

File: .github/workflows/node.js.yml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+      uses: actions/setup-node@v1
+      with:
```
Comment (charlotte-zhuang) [inline .github/workflows/node.js.yml:20]:
```suggestion
```

File: .github/workflows/node.js.yml [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+        node-version: ${{ matrix.node }}
```
Comment (charlotte-zhuang) [inline .github/workflows/node.js.yml:21]:
```suggestion
```

**Review by charlotte-zhuang — CHANGES_REQUESTED** [review]
`matrix.node` is undefined.

**Review by charlotte-zhuang — CHANGES_REQUESTED** [review]
`Run npm run frontend-tests` seems to have several connection refused errors. I couldn't find where they were being generated exactly. Can someone check on that? None of the tests are actually failing though.

Don't commit `package.json` and `package-lock.json`

**Review by IsitaB — CHANGES_REQUESTED** [review]
Please revert api/package-lock, package-lock

and ./package.json

Everything else works great!

**Review by Christopher-Tom — COMMENTED** [review]
yis, charlotte helped me with this

**Review by Christopher-Tom — COMMENTED** [review]
okay

**Review by charlotte-zhuang — APPROVED** [review]
`package.json` and `package-lock.json` changes are necessary or tests will fail. The last two commits aren't critical.

**Pull request merged**